### PR TITLE
feat: Add typerighter enabled command

### PR DESCRIPTION
**Pull request by rhystmills**
This PR adds a command to enable or disable Typerighter. This will be initially used in the liveblog, where Typerighter will be enabled (or disabled) via a toggle, after which realtime checks will occur (or stop occurring).

When Typerighter is enabled: 
- a check occurs of the whole document.
- realtime checks will continue to occur if they are enabled.

When Typerighter is disabled:
- all matches are removed
- all pending requests are discarded
- realtime checks will no longer occur

#### Manual testing
1. Run this repository locally according to the instructions in the readme.
2. Try disabling and enabling Typerighter via the toggle in the local application's UI. Are all of the following true:?
     - matches are removed from the UI when Typerighter is disabled.
     - no pending matches appear after Typerighter is disabled.
     - a check occurs when Typerighter is re-enabled
     - real time checks continue to occur when Typerighter is enabled and changes are made to the document
#### Automated testing
3. All tests can be run using `npm run test`. Are the tests added by this PR (and all others) passing?

### CODE REVIEW

File: src/ts/state/test/reducer.spec.ts [hunk 1]
```diff
@@ -839,6 +840,82 @@ describe("Action handlers", () => {
...
+          requestsInFlight: createRequestInFlight([createBlock(5, 10)]),
+          requestPending: true,
+        },
+        setTyperighterEnabled(false)
+      );
+
+      expect(newState.requestsInFlight).toEqual({});
+      expect(newState.requestPending).toEqual(false);
+    })
+    it("should fetch matches for the entire document when enabled", () => {
```
Comment (jonathonherbert) [inline src/ts/state/test/reducer.spec.ts:880]:
```suggestion
    it("should add requests-in-flight for the entire document when enabled", () => {
```

**Review by jonathonherbert — COMMENTED** [review]
A couple of minor comments, but basically good to 🚢 .

One thing I wrangled with – traditionally, action handlers are pure, and uuid generation isn't. Hence passing requestIds into actions elsewhere. But I don't think this is a hill worth dying on – in fact, it would make things much simpler everywhere we do it. So, I'm for it.

File: src/ts/commands.ts [hunk 2]
```diff
@@ -381,6 +382,24 @@ const maybeApplySuggestions = (
   return true;
 };
 
+/**
+ * Enable or disable typerighter
```
[the remark this answers is not in the document]
Comment (rhystmills) [reply src/ts/commands.ts:386]: Great idea.

File: src/ts/createTyperighterPlugin.ts [hunk 3]
```diff
@@ -134,7 +136,8 @@ const createTyperighterPlugin = <TFilterState, TMatch extends IMatch>(
     isElementPartOfTyperighterUI = () => false,
     requestMatchesOnDocModified = false,
     adapter,
-    telemetryAdapter
+    telemetryAdapter,
+    typerighterEnabled,
```
[the remark this answers is not in the document]
Comment (rhystmills) [reply src/ts/createTyperighterPlugin.ts]: Now done.

File: src/ts/state/test/store.spec.ts [hunk 4]
```diff
@@ -4,7 +4,7 @@ import { createDoc, p } from "../../test/helpers/prosemirror";
 
 describe("store", () => {
   const doc = createDoc(p("Example doc"))
-  const initState = createInitialState({ doc });
+  const initState = createInitialState({ doc, typerighterEnabled: true });
```
[the remark this answers is not in the document]
Comment (rhystmills) [reply src/ts/state/test/store.spec.ts]: Makes sense! Have done so now.

File: src/ts/test/helpers/createEditor.ts [hunk 5]
```diff
@@ -34,7 +34,8 @@ export const createEditor = (htmlDoc: string, matches: IMatch[] = []) => {
   const { plugin: validatorPlugin, getState } = createTyperighterPlugin({
     isElementPartOfTyperighterUI,
     matches,
-    adapter: new TyperighterAdapter("https://checker.typerighter.local.dev-gutools.co.uk")
+    adapter: new TyperighterAdapter("https://checker.typerighter.local.dev-gutools.co.uk"),
+    typerighterEnabled: true
```
[the remark this answers is not in the document]
Comment (rhystmills) [reply src/ts/test/helpers/createEditor.ts:38]: Now removed.

File: src/ts/state/test/reducer.spec.ts (hunk 1 above)
Comment (rhystmills) [reply src/ts/state/test/reducer.spec.ts]: Now done!

File: src/ts/state/reducer.ts [hunk 6]
```diff
@@ -840,3 +848,28 @@ const handleSetFilterState = <TPluginState extends IPluginState>(
...
+
+const handleSetTyperighterEnabled = (getIgnoredRanges: TGetSkippedRanges = doNotSkipRanges) => <TPluginState extends IPluginState>(
+  _: Transaction,
+  state: TPluginState,
+  { payload: { typerighterEnabled } }: ActionSetTyperighterEnabled,
+): TPluginState => {
+
+  const matchRequestAction = { type: REQUEST_FOR_DOCUMENT, payload: { requestId: v4(), categoryIds: []}}
+  const updatedState = typerighterEnabled ? {
+    // Run a check if typerighter has been enabled
```
Comment (jonathonherbert) [inline src/ts/state/reducer.ts:860]: Non-blocking – we should not need to spread our state into a new object, as you will always get a new identity from an action handler.

**Review by jonathonherbert — APPROVED** [review]

**Pull request merged**